quickwit/quickwit-indexing/src/actors/merge_planner.rs

```diff
@@ -331,70 +275,24 @@ impl MergePlanner {
         Ok(merge_operations)
     }
 
-    fn num_young_splits(&self) -> usize {
-        self.partitioned_young_splits
-            .values()
-            .map(|splits| splits.len())
-            .sum()
-    }
-
     async fn send_merge_ops(&mut self, ctx: &ActorContext<Self>) -> Result<(), ActorExitStatus> {
-        // We do not want to simply schedule all available merge operations here.
+        // We identify all of the merge operations we want to run and leave it
+        // to the merge scheduler to decide in which order these should be scheduled.
         //
-        // The reason is that in presence of partitioning, it is very possible
-        // to receive a set of splits opening the opportunity to run a lot "large" merge
-        // operations at the same time.
-        //
-        // These large merge operation will in turn prevent small merge operations
-        // from being executed, when in fact small merge operations should be executed
-        // in priority.
-        //
-        // As an alternative approach, this function push merge operations until it starts
-        // experience some push back, and then just "loops".
-        let mut merge_ops = self.compute_merge_ops(ctx).await?;
-        // We run smaller merges in priority.
-        merge_ops.sort_by_cached_key(|merge_op| Reverse(max_merge_ops(merge_op)));
-        while let Some(merge_operation) = merge_ops.pop() {
-            info!(merge_operation=?merge_operation, "planned merge operation");
+        // The merge scheduler has the merit of knowing about merge operations from other
+        // index as well.
+        let merge_ops = self.compute_merge_ops(ctx).await?;
+        for merge_operation in merge_ops {
+            info!(merge_operation=?merge_operation, "schedule merge operation");
```

Comment:
Maybe log one time for all ops?

Reply:
i actually prefer one at a time. It makes it easier to count etc. and this is not a very verbose log anyway.